rpc/src/v1/impls/eth.rs

```diff
@@ -339,7 +340,10 @@ impl<C, SN: ?Sized, S: ?Sized, M, EM> Eth for EthClient<C, SN, S, M, EM> where
 
 		let store = take_weak!(self.accounts);
 		let accounts = try!(store.accounts().map_err(|e| errors::internal("Could not fetch accounts.", e)));
-		Ok(accounts.into_iter().map(Into::into).collect())
+		let addresses = try!(store.accounts_info().map_err(|e| errors::internal("Could not fetch accounts.", e)));
```

Comment:
`accounts_info` should always be a superset of `accounts`, so there should be no need to merge the two.